miner/worker.go

```diff
@@ -911,16 +914,20 @@ func (w *worker) commitTransaction(tx *types.Transaction, coinbase common.Addres
 		// 2.1 when starting handling the first tx, `state.refund` is 0 by default,
 		// 2.2 after tracing, the state is either committed in `core.ApplyTransaction`, or reverted, so the `state.refund` can be cleared,
 		// 2.3 when starting handling the following txs, `state.refund` comes as 0
-		traces, err = w.current.traceEnv.GetBlockTrace(
-			types.NewBlockWithHeader(w.current.header).WithBody([]*types.Transaction{tx}, nil),
-		)
+		withTimer(l2CommitTraceTimer, func() {
+			traces, err = w.current.traceEnv.GetBlockTrace(
+				types.NewBlockWithHeader(w.current.header).WithBody([]*types.Transaction{tx}, nil),
+			)
+		})
 		// `w.current.traceEnv.State` & `w.current.state` share a same pointer to the state, so only need to revert `w.current.state`
 		// revert to snapshot for calling `core.ApplyMessage` again, (both `traceEnv.GetBlockTrace` & `core.ApplyTransaction` will call `core.ApplyMessage`)
 		w.current.state.RevertToSnapshot(snap)
 		if err != nil {
 			return nil, nil, err
 		}
-		accRows, err = w.circuitCapacityChecker.ApplyTransaction(traces)
+		withTimer(l2CommitCCCTimer, func() {
+			accRows, err = w.circuitCapacityChecker.ApplyTransaction(traces)
+		})
 		if err != nil {
```

Comment:
Should we also time `core.ApplyTransaction`?

Reply:
probably